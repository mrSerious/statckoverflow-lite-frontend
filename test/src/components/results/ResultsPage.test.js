import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { store } from '../../../../src/store/store';
import Results, { ResultPage } from '../../../../src/components/results/ResultsPage';

const props = {
  results: ['test'],
  history: { push: jest.fn() },
};

describe('results page tests', () => {
  const wrapper = shallow(<ResultPage {...props} />);

  it('should render the results page without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Results mount tests', () => {
  const wrapper = mount(
    <MemoryRouter>
      <Results store={store} {...props} />
    </MemoryRouter>
  );
  const instance = wrapper.find('ResultPage').instance();

  it('mapStateToProps works as expected', () => {
    expect(instance.props.results).toBeDefined();
  });
});
