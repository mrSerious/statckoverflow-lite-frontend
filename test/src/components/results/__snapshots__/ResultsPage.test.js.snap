// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should render the results page without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
